File: BlackJack-CSHARP/Blackjack21.Game/Game/Model/Card.cs
namespace Blackjack21.Game.Model
{
    /// <summary>
    /// Blackjack Card Object
    /// </summary>
    public class Card
    {
        private CardType _cardType;
        private CardSymbol _cardSymbol;
        private int _cardValue = 0;

        /// <summary>
        /// Initializes the card object
        /// </summary>
        /// <param name="cardType">Set Card Type</param>
        /// <param name="cardSymbol">Set Card Symbol</param>
        public Card(CardType cardType = CardType.ACE, CardSymbol cardSymbol = CardSymbol.CLUBS)
        {
            SetCard(cardType, cardSymbol);
        }

        /// <summary>
        /// initialize the card based on number parameters inputs
        /// </summary>
        /// <param name="cardType">Set Card Type</param>
        /// <param name="cardSymbol">Set Card Color</param>
        public Card(int cardType, int cardSymbol)
        {
            SetCard((CardType)cardType, (CardSymbol)cardSymbol);
        }

        /// <summary>
        /// Set cards identity and calulate the value of the card
        /// </summary>
        /// <param name="cardType"></param>
        /// <param name="cardSymbol"></param>
        private void SetCard(CardType cardType, CardSymbol cardSymbol)
        {
            this._cardSymbol = cardSymbol;
            this._cardType = cardType;
            SetCardValue();
        }

        /// <summary>
        /// Gets card's type
        /// </summary>
        public CardType CardType
        {
            get
            {
                return this._cardType;
            }
        }

        /// <summary>
        /// Gets card's symbol
        /// </summary>
        public CardSymbol CardSymbol
        {
            get
            {
                return this._cardSymbol;
            }
        }

        /// <summary>
        /// Return card's number value
        /// </summary>
        public int CardValue
        {
            get
            {
                return this._cardValue;
            }
        }

        /// <summary>
        /// Compare the card type against the another card
        /// </summary>
        /// <param name="card">Card to compare with</param>
        /// <returns>Returns if the card type is the same as the other card type</returns>
        public bool CompareCardType(Card card)
        {
            return card != null && card.CardType == this.CardType;
        }

        /// <summary>
        /// Compare the card value against the another card
        /// </summary>
        /// <param name="card">Card to compare with</param>
        /// <returns>Returns if the card value is the same as the other card value</returns>
        public bool CompareCardValue(Card card)
        {
            return card != null && card.CardValue == this.CardValue;
        }

        /// <summary>
        /// Set card's value
        /// </summary>
        private void SetCardValue()
        {
            int cardTypeInt = (int)this._cardType;

            // if the card type value is over 10 count it as 10
            this._cardValue = cardTypeInt > 9 ? 10 : cardTypeInt;
        }

        /// <summary>
        /// Returns the card name as a string value
        /// </summary>
        /// <returns>returns the string name of the card</returns>
        public override string ToString()
        {
            return this.CardType + " OF " + this._cardSymbol.ToString(); ;
        }

    }
}

File: Blackjack21-JAVA/Blackjack21Library/src/Blackjack21/Game/Model/Card.java
package Blackjack21.Game.Model;

/**
 * Blackjack Card Object
 */
public class Card
{

    private CardType _cardType = CardType.ACE;
    private CardSymbol _cardSymbol = CardSymbol.CLUBS;
    private int _cardValue = 0;

    /**
     * Initialize the card based on input parameters
     *
     * @param cardType Set Card Type
     * @param cardSymbol Set Card Symbol
     */
    public Card(CardType cardType, CardSymbol cardSymbol)
    {
        setCard(cardType, cardSymbol);
    }

    /**
     * initialize the card based on number parameters inputs
     *
     * @param cardType Set Card Type
     * @param cardSymbol Set Card Color
     */
    public Card(int cardType, int cardSymbol)
    {
        setCard(CardType.values()[cardType], CardSymbol.values()[cardSymbol]);
    }

    /**
     * Set cards identity and calculate the value of the card
     *
     * @param cardType
     * @param cardSymbol
     */
    private void setCard(CardType cardType, CardSymbol cardSymbol)
    {
        this._cardSymbol = cardSymbol;
        this._cardType = cardType;
        setCardValue();
    }

    /**
     * Sets or gets card's type
     *
     * @return return the card's type
     */
    public CardType getCardType()
    {
        return this._cardType;
    }

    /**
     * Gets card's symbol
     *
     * @return return the card's symbol
     */
    public CardSymbol getCardSymbol()
    {
        return this._cardSymbol;
    }

    /**
     * Return card's number value
     *
     * @return return the cards value
     */
    public int getCardValue()
    {
        return this._cardValue;
    }

    /**
     * Compare if the card is the same type as this card
     *
     * @param card Card to compare with
     * @return return the result of the compare
     */
    public boolean compareCardType(Card card)
    {
        return card != null && card.getCardType() == this.getCardType();
    }

    /**
     * Set card's value
     */
    private void setCardValue()
    {
        int cardTypeInt = ((Enum) this._cardType).ordinal();
        // if the card type value is over 10 count it as 10
        this._cardValue = cardTypeInt > 9 ? 10 : cardTypeInt;
    }

    /**
     * Returns the card as a string value
     *
     * @return return the string value of the card
     */
    @Override
    public String toString()
    {

        String cardName = "";
        switch (this.getCardType())
        {
            case ACE:
                cardName = "ACE OF ";
                break;
            case TWO:
                cardName = "TWO OF ";
                break;
            case THREE:
                cardName = "THREE OF ";
                break;
            case FOUR:
                cardName = "FOUR OF ";
                break;
            case FIVE:
                cardName = "FIVE OF ";
                break;
            case SIX:
                cardName = "SIX OF ";
                break;
            case SEVEN:
                cardName = "SEVEN OF ";
                break;
            case EIGHT:
                cardName = "EIGHT OF ";
                break;
            case NINE:
                cardName = "NINE OF ";
                break;
            case TEN:
                cardName = "TEN OF ";
                break;
            case JACK:
                cardName = "JACK OF ";
                break;
            case QUEEN:
                cardName = "QUEEN OF ";
                break;
            case KING:
                cardName = "KING OF ";
                break;

        }
        switch (this._cardSymbol)
        {
            case CLUBS:
                cardName += "CLUBS";
                break;
            case SPADES:
                cardName += "SPADES";
                break;
            case HEARTS:
                cardName += "HEARTS";
                break;
            case DIAMONDS:
                cardName += "DIAMONDS";
                break;

        }
        return cardName;

    }

}
